LFPy/eegmegcalc.py

```diff
@@ -0,0 +1,476 @@
+            the x-, y- and z-direction.
+
+        Returns
+        _______
+        potential : ndarray [nV]
+            Array containing the current dipole moment at point r.
+        """
+
+        p_tot = np.linalg.norm(p_rad, axis=1)
+        theta = self.calc_theta()
+        s_vector = self.sign_rad_dipole(p_rad)
+        phi_const = s_vector * p_tot / (4 * np.pi * self.sigma1 * self.rz ** 2) * self.k1
+        n_terms = np.zeros((len(self.r), len(p_tot)))
+        for el_point in range(len(self.r)):
+            el_rad = self.r[el_point]
+            theta_point = theta[el_point]
+            if el_rad <= self.r1:
+                n_terms[el_point] = self.potential_brain_rad(el_rad, theta_point)
+            elif el_rad <= self.r2:
+                n_terms[el_point] = self.potential_csf_rad(el_rad, theta_point)
+            elif el_rad <= self.r3:
+                n_terms[el_point] = self.potential_skull_rad(el_rad, theta_point)
+            elif el_rad <= (self.r4):
+                n_terms[el_point] = self.potential_scalp_rad(el_rad, theta_point)
+            elif el_rad <= (self.r4+1E-6):
+                el_rad = self.r4
+                n_terms[el_point] = self.potential_scalp_rad(el_rad, theta_point)
+            else:
+                n_terms[el_point] = np.nan
+                raise ValueError('Electrode located outside head model. Maximum r = %s µm.', self.r4, '\n your r = ', self.r)
+        potential = phi_const * n_terms
+        return potential
+
+    def calc_tan_potential(self, p_tan):
+        """Return potential from tangential dipole P at location rz measured at r
+        Parameters
+        __________
+        p : ndarray [1E-15 Am]
+            Array containing the current dipole moment for all timesteps in
+            the x-, y- and z-direction.
+
+        Returns
+        _______
+        potential : ndarray [nV]
+            Array containing the current dipole moment at point r.
+        """
+        theta = self.calc_theta()
+        phi = self.calc_phi(p_tan)
+        p_tot = np.linalg.norm(p_tan, axis=1)
+        phi_hom = - p_tot / (4 * np.pi * self.sigma1 * self.rz ** 2) * np.sin(phi) * self.k1
+        n_terms = np.zeros((len(self.r), len(p_tot)))
+        for el_point in range(len(self.r)):
+            el_rad = self.r[el_point]
+            theta_point = theta[el_point]
+            if el_rad <= self.r1:
+                n_terms[el_point] = self.potential_brain_tan(el_rad, theta_point)
+            elif el_rad <= self.r2:
+                n_terms[el_point] = self.potential_csf_tan(el_rad, theta_point)
+            elif el_rad <= self.r3:
+                n_terms[el_point] = self.potential_skull_tan(el_rad, theta_point)
+            elif el_rad <= self.r4:
+                n_terms[el_point] = self.potential_scalp_tan(el_rad, theta_point)
+            else:
+                n_terms[el_point] = np.nan
+        potential = phi_hom * n_terms
+        return potential
+
+    def calc_theta(self):
```

Comment:
I think some of these class methods are probably only meant to be used internally by the above methods. If so, can you make them non-public, i.e., precede the method names by an underscore?